modules/stanford_soe_helper_homepage/stanford_soe_helper_homepage.module

```diff
@@ -5,3 +5,24 @@
  */
 
 include_once 'stanford_soe_helper_homepage.features.inc';
+
+/**
+ * Implementation of hook_init().
+ */
+function stanford_soe_helper_homepage_init() {
```

Comment:
Yes, hook_install is a more appropriate place for this code. 
Please add a stanford_soe_helper_homepage.install file and the hook_install.